File: navigation_stack_saved_spot/my_turtlebot_navigation/src/get_coordinates_service_server.py
#!/usr/bin/env python3
import yaml
import rospy
from my_turtlebot_navigation.srv import MyCustomServiceMessage, MyCustomServiceMessageResponse
from geometry_msgs.msg import Pose

file_path = "/home/rezaeifh/catkin_ws/src/Turtlebot3-Navigation/navigation_stack_saved_spot/my_turtlebot_localization/config/spots.yaml"


class Spotfind():

    def __init__(self):

        # Start a service to find the goal pose
        rospy.Service('/get_coordinates', MyCustomServiceMessage,
                      self.service_callback)

    # Service callback function
    def service_callback(self, request):

        my_response = MyCustomServiceMessageResponse()

        # It finds the label and separates corresponding position
        with open(file_path, "r") as f:
            docs = yaml.load_all(f, Loader=yaml.FullLoader)
            for x in docs:
                for k, v in x.items():
                    if k == request.label:
                        rospy.loginfo("getting position")
                        my_response.goalpose = v

        my_response.message = ("OK")
        rospy.loginfo(my_response.message)

        return my_response


if __name__ == "__main__":
    rospy.init_node("spot_finder")
    Spotfind()
    rospy.spin()

File: navigation_stack_saved_spot/my_turtlebot_navigation/src/get_coordinates_service_server.cpp
#include <ros/ros.h>
#include <geometry_msgs/Pose.h>
#include <my_turtlebot_navigation/MyCustomServiceMessage.h>
#include <sensor_msgs/LaserScan.h>
#include <string>


class Spotfind
{
protected:
    
    std::string file_path = "/home/rezaeifh/catkin_ws/src/Turtlebot3-Navigation/navigation_stack_saved_spot/my_turtlebot_localization/config/spots.yaml";
    
    // ROS Objects
    ros::NodeHandle nh_;

    std::string line;

public:    
    //Start a service to find the goal pose            
    ros::ServiceServer my_service = nh_.advertiseService("/get_coordinates", &Spotfind::service_callback, this);  
    
    // Service callback function
    bool service_callback(my_turtlebot_navigation::MyCustomServiceMessage::Request &request,
                         my_turtlebot_navigation::MyCustomServiceMessage::Response &my_response)
    {
        
        nh_.getParam(request.label, line);
        ROS_INFO("getting_position");
        my_response.goalpose = line;
        my_response.message = "OK";
        ROS_INFO("%s",my_response.message.c_str());

        return true;
    }
};
    
int main(int argc, char** argv)
{
  ros::init(argc, argv, "spot_finder");
  Spotfind Spotfind;
  ros::spin();
  
  return 0;
}
